### PR TITLE
Byt 12 ping pong front end

### PR DESCRIPTION
**Pull request by jackadrianglass**
Added a simple ping pong end point to get some experience with routing and http requests in gleam.

### CODE REVIEW

File: client/src/bc/landing_page.gleam [hunk 1]
```diff
@@ -1,23 +1,51 @@
...
+    ApiUpdatedPing(Ok(ping)) -> #(Model(ping: Some(ping)), effect.none())
+    ApiUpdatedPing(Error(_)) -> #(model, effect.none())
+  }
 }
 
-pub fn update(model, _msg) {
-  model
+fn ping() -> Effect(Msg) {
+  // fixme: use the real url here
+  let url = "http://localhost:8000/ping"
```
Comment (jackadrianglass) [inline client/src/bc/landing_page.gleam]: This shouldn't be in the deployed application. Any suggestions on what we could do for the testing vs actual prod url should be? Maybe a command line argument?

Comment (lakinwecker) [reply client/src/bc/landing_page.gleam]: Usually, you use an env variable, but in the case of a pre-built set of javascript served to the browser, there aren't any env variables, so you usually default to the schema/domain/port that the app was served on.  In general, it's actually best if this is the way it's also served locally as well

Comment (lakinwecker) [reply client/src/bc/landing_page.gleam]: I don't know how to get access to the current scheme/domain/port in gleam, but you can certainly get access to it in Javascript, so it should be possible.

In Elm, I think the recommended way to do things like this was to have the javascript that does the initial page load generate this kind of thing and then pass it in on init.

File: server/src/bc/router.gleam [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+
+fn home_page(req: Request, ctx: web.Context) -> Response {
+  use <- wisp.require_method(req, Get)
   // Default to the index.html if nothing else is here.
   response(200)
   |> set_body(File(debug(ctx.static_directory <> "/index.html")))
 }
+
+fn pong(req: Request) -> Response {
+  use <- wisp.require_method(req, Get)
```
Comment (lakinwecker) [inline server/src/bc/router.gleam]: This maybe should be a POST

**Review by lakinwecker — APPROVED** [review]
Provided we fix the url stuff, this looks good to me

File: client/src/bc/landing_page.gleam (hunk 1 above)
Comment (jackadrianglass) [reply client/src/bc/landing_page.gleam]: Alright I think I came up with a solution for this. Let me know what you think

**Pull request merged**